tidy3d/components/data/monitor_data.py

```diff
@@ -996,6 +1009,111 @@ def to_source(
             field_dataset=dataset, source_time=source_time, center=center, size=size, **kwargs
         )
 
+    def make_adjoint_sources(
+        self, dataset_names: list[str]
+    ) -> List[Union[CustomCurrentSource, PointDipole]]:
+        """Converts a :class:`.FieldData` to a list of adjoint current or point sources."""
+
+        if np.allclose(np.array(self.monitor.size), np.zeros(3)):
```

Comment:
This works without `np.zeros()` and casting, i.e. `np.allclose(self.monitor.size, 0)`

Reply:
thanks